packages/client/Atmosphere.ts

```diff
@@ -437,4 +446,11 @@ export default class Atmosphere extends Environment {
     this.viewerId = null!
     this.userId = null // DEPRECATED
   }
+
+  private resetAuth() {
+    this.authToken = null
+    this.authObj = null
+    window.localStorage.removeItem(LocalStorageKey.APP_TOKEN_KEY)
+    window.localStorage.removeItem(LocalStorageKey.EMAIL)
```

Comment:
Thanks for this context. What do you think about caching the entire `ViewerInfo` result in LocalStorage? Then, on load, we could default the `viewerInfo` state to the values in LocalStorage. I think that would satisfy the "identify with Segment ASAP" requirement. 

Additionally, this would identify with all the other third parties faster as well.

Related to your comment below (https://github.com/ParabolInc/parabol/pull/5991#discussion_r813272371), if we're not concerned about the extra traffic, we could use the values from LocalStorage and also trigger a refresh against the endpoint on load. Then, if there are updates, we'd re-identify with the updated info early-on in the app lifecycle. This behavior would be similar to Apollo's `cache-and-network` [fetchPolicy](https://www.apollographql.com/docs/react/data/queries/#supported-fetch-policies).

Reply:
we've gotta keep this out of the relay cache entirely since the only new field is email (the userId is persisted in the JWT) and we want to access the email before a fetch & doing that within relay means persisting the object in the local storage, rehydrating a relay store & validating it